Implement methods in lateral class

MarekOchocki opened the pull request:
- penumbra (left and right)
- field_ratio
- symmetry
- flatness_50
- flatness_90
- asymmetry
- normalize
- absolute_y

MarekOchocki committed:
Implement methods in lateral class

- penumbra (left and right)
- field_ratio
- symmetry
- flatness_50
- flatness_90
- asymmetry
- normalize
- absolute_y

grzanka commented on ibodata/lateral_profile.py:
```diff
@@ -1,6 +1,63 @@
+import numpy as np
+import math
```
Sort imports correctly. See https://www.python.org/dev/peps/pep-0008/#imports

```
Imports should be grouped in the following order:

standard library imports
related third party imports
local application/library specific imports
You should put a blank line between each group of imports.
```

grzanka commented on ibodata/lateral_profile.py:
```diff
@@ -1,6 +1,63 @@
 class LateralProfile(Profile):
-    """TODO"""
-    pass
+    def penumbra_left(self):
```
Add a docstring explaining what this method returns: 
 - in case of good data
 - in case of corrupted data (i.e. only half of the profile is recorded)

grzanka commented on ibodata/lateral_profile.py:
```diff
@@ -1,6 +1,63 @@
+
+        self.y /= ave
+
+    def absolute_y(self):
```
Add a docstring explaining what this method does.

grzanka commented:
@MarekOchocki see also this one:
https://travis-ci.org/grzanka/ibodata/jobs/267213793#L467
(can you access it ?)

It means code doesn't follow PEP8 codestyle rules.
Fix it manually or use https://github.com/google/yapf

MarekOchocki committed:
Add comments explaining methods

- docstring explaining penumbra
- docstring explaining absolute_y
- imports sorted

grzanka commented on ibodata/lateral_profile.py:
```diff
@@ -1,6 +1,75 @@
+
+    def absolute_y(self):
+        """
+        Translate y to bring y.min() to 0 (removes noise)
```
More commonly used is "noise subtraction" than "noise removal".

grzanka commented on ibodata/lateral_profile.py:
```diff
@@ -1,6 +1,75 @@
+        """
+        Translate y to bring y.min() to 0 (removes noise)
+        """
+        self.y = self.y - self.y.min()
```
Why not `self.y -= ...` ?

grzanka commented on ibodata/lateral_profile.py:
```diff
@@ -1,6 +1,75 @@
+        Normalize to 1 over [-dt, +dt] area from the mid of the profile
+        """
+        a = self.y.max() - self.y.min()
+        a /= 2
```
Why not be more explicit and divide by 2.0

Float and integer division is an interesting issue in python2 and 3
See http://www.informit.com/articles/article.aspx?p=1439189 and https://www.python.org/dev/peps/pep-0238/

grzanka commented:
Could you also please implement some tests in https://github.com/grzanka/ibodata/blob/master/tests/lateral.py ?

MarekOchocki committed:
Replace constant integers with floats

MarekOchocki committed:
Merge absolute_y with normalize

You have to call absolute_y before calling normalize anyway.
If you call these in opposite way you'd have to call normalize again.

MarekOchocki committed:
Cosmetic changes to fit PEP8 style

MarekOchocki committed:
Cosmetic changes part 2
